File: src/main/java/systems/conduit/main/core/plugin/config/ConfigurationLoader.java
package systems.conduit.main.core.plugin.config;

import java.io.File;
import java.util.Optional;

/*
 * @author Innectic
 * @since 10/16/2019
 */
public interface ConfigurationLoader {

    Optional<Configuration> load(File file, Class<? extends Configuration> configurationType);
}

File: src/main/kotlin/systems/conduit/core/plugin/config/ConfigurationLoader.kt
package systems.conduit.core.plugin.config

import java.io.File
import java.util.*

/*
 * @author Innectic
 * @since 10/16/2019
 */
interface ConfigurationLoader {
    fun load(file: File, configurationType: Class<out Configuration>): Configuration?
}
